gbdxtools/auth.py

```diff
@@ -42,9 +42,39 @@ def __init__(self, **kwargs):
         except Exception as err:
             print(err)
 
+        def expire_token(r, *args, **kwargs):
+            """
+            Requests a new token if 401, retries request, mainly for auth v2 migration
+            :param r:
+            :param args:
+            :param kwargs:
+            :return:
+            """
+            if r.status_code == 401:
+                try:
+                    # remove hooks so it doesn't get into infinite loop
+                    r.request.hooks = None
+                    # expire the token
+                    gbdx_auth.expire_token(token_to_expire=self.gbdx_connection.token,
+                                           config_file=kwargs.get('config_file'))
+                    # re-init the session
+                    self.gbdx_connection = gbdx_auth.get_session(kwargs.get('config_file'))
```

Comment:
I need to go look at gbdx_auth again but this will work w/o a config_file as well?

Reply:
ok so i just looked at gbdx_auth.get_session and this will work, except i think you need to default this to `None`

`kwargs.get('config_file', None)` right?